### PR TITLE
Added the possibility of building and running the project on Mac (without bluetooth)

### PR DESCRIPTION
**Pull request by RamonBeast**
Due to bluepy not working on Mac, it's not possible to install and run the project on Mac devices as it is. I modified the project so that bluepy is excluded from the build if the platform is a Mac. I've also made some changes to basic.py to allow the main example to run on Mac as well.

### CODE REVIEW

**Comment by ckuethe** [comment]
Doesn't break on my Ubuntu machine.

File: radiacode-examples/basic.py [hunk 1]
```diff
@@ -1,20 +1,40 @@
...
+
+    parser.add_argument('--serial', type=str, required=False,
+                        help='serial number of radiascan device (e.g. "RC-10x-xxxxxx"). Useful in case of multiple devices.')
+
     args = parser.parse_args()
 
-    if args.bluetooth_mac:
-        print('will use Bluetooth connection')
-        rc = RadiaCode(bluetooth_mac=args.bluetooth_mac)
+    if hasattr(args, 'bluetooth_mac') and getattr(args, 'bluetooth_mac', None):
```
Comment (cdump) [inline radiacode-examples/basic.py:19]: `and getattr(args, 'bluetooth_mac', None):` can be simplified to `and args.bluetooth_mac:`

File: radiacode/radiacode.py [hunk 2]
```diff
@@ -25,7 +25,11 @@ def __init__(
         ignore_firmware_compatibility_check: bool = False,
     ):
         self._seq = 0
-        if bluetooth_mac is not None:
+
+        # Bluepy doesn't support MacOS: https://github.com/IanHarvey/bluepy/issues/44
+        self._bt_supported = False if platform.system() == 'Darwin' else True
```
Comment (cdump) [inline radiacode/radiacode.py:30]: `self._bt_supported = platform.system() != 'Darwin'`

**Review by cdump — CHANGES_REQUESTED** [review]
Thanks for your PR, I have two suggestions for code style

**Comment by cdump** [comment]
+ codestyle errors from tests: https://github.com/cdump/radiacode/actions/runs/8958200767/job/24765854405?pr=32

File: radiacode-examples/basic.py (hunk 1 above)
Comment (RamonBeast) [reply radiacode-examples/basic.py]: Change applied and pushed in the branch

File: radiacode/radiacode.py (hunk 2 above)
Comment (RamonBeast) [reply radiacode/radiacode.py]: Change applied and pushed in the branch

**Comment by cdump** [comment]
Please fix `ruff` linter errors from failing CI test (https://github.com/cdump/radiacode/actions/runs/9030194133/job/24818308952?pr=32) and I'll merge this

**Comment by RamonBeast** [comment]
I somehow missed it, fixed and pushed.

**Comment by cdump** [comment]
Still have code-format ruff issues - you can run tests localy: https://github.com/cdump/radiacode/blob/master/.github/workflows/tests.yml#L26

**Comment by RamonBeast** [comment]
Last one should be the good one!

**Pull request merged**